samples/snippets/get_metadata.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Google Analytics Data API sample application retrieving dimension and metrics
+metadata.
+"""
+from google.analytics.data_v1beta import BetaAnalyticsDataClient
+from google.analytics.data_v1beta.types import GetMetadataRequest
+from google.analytics.data_v1beta.types import MetricType
+
+
+def get_metadata_by_property_id(property_id="YOUR-GA4-PROPERTY-ID"):
+    """Retrieves dimensions and metrics available for a Google Analytics 4
+    property, including custom fields."""
+    client = BetaAnalyticsDataClient()
+
+    # [START analyticsdata_get_metadata_by_property_id]
```

Comment:
Kurtis, we do follow this advice for quickstart samples. However, other snippets are included in the API usage guides that demonstrate the usage of various API features inside one document (e.g. "Advanced API Features Guide", "Migration Guide"). It is therefore useful to include just the relevant part of the snippet to illustrate the feature. If a user wants to see the complete example, they can follow a link to GitHub.

Reply:
Here is an example of the guide we will need to include the samples into:
https://developers.devsite.corp.google.com/analytics/devguides/reporting/data/v1/basics

Reply:
Also, I grouped certain samples together in one file since they demonstrate the same concept (e.g. `run_report_with_filters.py`), just with different queries. Creating a separate file for each sample looks a bit redundant here, though we may consider separating these in the future. The more use cases we demonstrate, the easier it will be for the users to adopt the API...

Reply:
Yes, we should do it for each sample. The goal is that samples should be "copy-paste-runnable" - most users want to be able to copy a sample, paste it into their own environment, and run it immediately with only minor changes. This is what the sample format is optimizing for. 

It may seem redundant but having each sample isolated into its own file makes it easier to ensure only used imports are included and that each snippet make sense without additional context.

Reply:
I agree this makes sense. I moved each sample into a separate file and moved region tags around to include the entire sample.